src/control/zoom/ZoomControl.cpp

```diff
@@ -26,6 +26,33 @@ auto onScrolledwindowMainScrollEvent(GtkWidget* widget, GdkEventScroll* event, Z
     return zoom->isZoomPresentationMode();
 }
 
+auto onTouchpadPinchEvent(GtkWidget* widget, GdkEventTouchpadPinch* event, ZoomControl* zoom) -> bool {
+    if (event->type == GDK_TOUCHPAD_PINCH && event->n_fingers == 2) {
+        utl::Point<double> center;
+        switch (event->phase) {
+            case GDK_TOUCHPAD_GESTURE_PHASE_BEGIN:
+                if (zoom->isZoomFitMode()) {
+                    zoom->setZoomFitMode(false);
+                }
+                center = {event->x, event->y};
+                // Need to adjust the center because the event coordinates are relative
+                // to the widget, not to the zoomed (and translated) view
+                center += {zoom->getVisibleRect().x, zoom->getVisibleRect().y};
```

Comment:
So when doing the zooming for the touchscreen, this is wrapped in an if statement, checking whether "getGtkTouchscreenScrollingEnabled()" is false. In my testing, adding this offset was always necessary. With the touch work around enabled, the whole pinch gesture doesn't work at all though.

Reply:
You may keep it as it is then. `getGtkTouchscreenScrollingEnabled()` will be removed soon anyway, as this workaround got revamped into better internal handling of scrolling on touchscreens.